test/src/test/java/org/apache/accumulo/test/functional/TableChangeStateIT.java

```diff
@@ -0,0 +1,549 @@
+    log.debug("Time for online when offline {} ms", TimeUnit.MILLISECONDS.convert(status2.runningTime(), TimeUnit.NANOSECONDS));
+    log.debug("Time for blocked online {} ms", TimeUnit.MILLISECONDS.convert(status3.runningTime(), TimeUnit.NANOSECONDS));
+
+    compactThread.join();
+
+  }
+
+  /**
+   * Blocks current thread until compaction is running.
+   *
+   * @return true if compaction and associate fate found.
+   */
+  private boolean blockUntilCompactionRunning() {
+
+    try {
+
+      Connector connector = getConnector();
+
+      int runningCompactions = 0;
+
+      List<String> tservers = connector.instanceOperations().getTabletServers();
+
+      /*
+       * wait for compaction to start - The compaction will acquire a fate transaction lock that used to block a subsequent online command while the fate
+       * transaction lock was held.
+       */
+      while (runningCompactions == 0) {
+
+        try {
+
+          for (String tserver : tservers) {
+            runningCompactions += connector.instanceOperations().getActiveCompactions(tserver).size();
+            log.trace("tserver {}, running compactions {}", tservers, runningCompactions);
+          }
+
+        } catch (AccumuloSecurityException | AccumuloException ex) {
+          throw new IllegalStateException("failed to get active compactions, test fails.", ex);
+        }
+
+        try {
+          Thread.sleep(250);
+        } catch (InterruptedException ex) {
+          // reassert interrupt
+          Thread.currentThread().interrupt();
+        }
+      }
+
+      // Validate that there is a compaction fate transaction - otherwise test is invalid.
+      return findFate();
+
+    } catch (AccumuloSecurityException | AccumuloException ex) {
+      throw new IllegalStateException("test failed waiting for compaction to start", ex);
+    }
+  }
+
+  /**
+   * Checks fates in zookeeper looking for transaction associated with a compaction as a double check that the test will be valid because the running compaction
+   * does have a fate transaction lock.
+   *
+   * @return true if corresponding fate transaction found, false otherwise
+   * @throws AccumuloException
+   *           from {@link ConfigurableMacIT#getConnector}
+   * @throws AccumuloSecurityException
+   *           from {@link ConfigurableMacIT#getConnector}
+   */
+  private boolean findFate() throws AccumuloException, AccumuloSecurityException {
+
+    Connector connector = getConnector();
+
+    String zPath = ZooUtil.getRoot(connector.getInstance()) + Constants.ZFATE;
+    ZooReader zooReader = new ZooReader(connector.getInstance().getZooKeepers(), connector.getInstance().getZooKeepersSessionTimeOut());
+    ZooCache zooCache = new ZooCache(zooReader, null);
+
+    List<String> lockedIds = zooCache.getChildren(zPath);
+
+    for (String fateId : lockedIds) {
+
+      List<String> lockNodes = zooCache.getChildren(zPath + "/" + fateId);
```

Comment:
This code assumes fate internals work a certain way.  If fate internals a changed, then someone would need to know to change this method OR we need to be 100% sure this will fail if internals are changed.  I was wondering if internal FATE APIs could be used here.  I looked at ZooStore and I am not sure if has everything you need.   I am wondering if the test would be ok w/o this method?